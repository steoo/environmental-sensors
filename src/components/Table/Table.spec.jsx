import React from 'react';
import { render } from '@testing-library/react';
import ResultsTable from './Table.component';

const tableData = {
  results: [
    {
      id: 'Box-A1-O3',
      box_id: 'Box-A1',
      sensor_type: 'O3',
      unit: 'ppm',
      name: 'Ozone',
      range_l: 0.0,
      range_u: 1000.0,
      longitude: -0.06507,
      latitude: 51.51885,
      reading: 672,
      reading_ts: '2019-09-10T00:00:00',
    },
    {
      id: 'Box-A1-NO2',
      box_id: 'Box-A1',
      sensor_type: 'NO2',
      unit: 'ppm',
      name: 'Nitrogen dioxide',
      range_l: 0.0,
      range_u: 1000.0,
      longitude: -0.06507,
      latitude: 51.51885,
      reading: 193,
      reading_ts: '2019-09-10T00:00:00',
    },
    {
      id: 'Box-A1-CO',
      box_id: 'Box-A1',
      sensor_type: 'CO',
      unit: 'ppm',
      name: 'Carbon monoxide',
      range_l: 0.0,
      range_u: 1000.0,
      longitude: -0.06507,
      latitude: 51.51885,
      reading: 331,
      reading_ts: '2019-09-10T00:00:00',
    },
  ],
};

describe('Components', () => {
  describe('Table', () => {
    it('should render, without crashing', () => {
      const { container } = render(<ResultsTable {...tableData} />);

      expect(container).toMatchSnapshot();
    });
  });
});
