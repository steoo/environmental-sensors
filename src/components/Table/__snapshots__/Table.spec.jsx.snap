// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Table should render, without crashing 1`] = `
<div>
  <div
    class="MuiTableContainer-root sc-bdfBwQ bSeEsD"
  >
    <table
      class="MuiTable-root MuiTable-stickyHeader"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
            scope="col"
          >
            ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
            scope="col"
          >
            Box ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
            scope="col"
          >
            Sensor Type
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
            scope="col"
          >
            Unit
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
            scope="col"
          >
            Reading
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
            scope="col"
          >
            Timestamp
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Box-A1-O3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Box-A1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            O3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Ozone
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            ppm
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            672
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            2019-09-10T00:00:00
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Box-A1-NO2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Box-A1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            NO2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Nitrogen dioxide
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            ppm
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            193
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            2019-09-10T00:00:00
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Box-A1-CO
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Box-A1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            CO
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Carbon monoxide
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            ppm
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            331
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            2019-09-10T00:00:00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
